packages/@aws-cdk/aws-elasticloadbalancingv2/lib/alb/application-listener-rule.ts

```diff
@@ -160,6 +171,214 @@ export interface RedirectResponse {
   readonly statusCode: string;
 }
 
+/**
+ * Interface for a listener rule condition
+ */
+export abstract class ListenerRuleCondition {
```

Comment:
Feel like this could be `ListenerCondition`, not as precise but I like shorter names.

Reply:
Please put the conditions in a separate file `conditions.ts`.